examples/step-89/step-89.cc

```diff
@@ -0,0 +1,1884 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2023 by the deal.II authors
+ *
+ * This file is part of the deal.II library.
+ *
+ * The deal.II library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE.md at
+ * the top level directory of deal.II.
+ *
+ * ---------------------------------------------------------------------
+ *
+ *
+ * Authors: Johannes Heinz, TU Wien, 2023
+ *          Marco Feder, SISSA, 2023
+ *          Peter Munch, University of Augsburg, 2023
+ */
+
+// @sect3{Include files}
+//
+// The program starts with including all the relevant header files.
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/mpi.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/mapping_q1.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/grid_tools_cache.h>
+
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <deal.II/non_matching/mapping_info.h>
+
+#include <deal.II/matrix_free/matrix_free.h>
+#include <deal.II/matrix_free/fe_evaluation.h>
+#include <deal.II/matrix_free/operators.h>
+// The following header file provides the class FERemoteEvaluation, which allows
+// to access values and/or gradients at remote triangulations similar to
+// FEEvaluation.
+#include <deal.II/matrix_free/fe_remote_evaluation.h>
+
+// We pack everything that is specific for this program into a namespace
+// of its own.
+namespace Step89
+{
+  using namespace dealii;
+
+  // @sect3{Initial conditions for vibrating membrane}
+  //
+  // Function that provides the initial condition for the vibrating membrane
+  // testcase.
+  template <int dim>
+  class InitialConditionVibratingMembrane : public Function<dim>
+  {
+  public:
+    InitialConditionVibratingMembrane(const double modes)
```

Comment:
We need to split up the the class declartions/definitions.

Reply:
[04cb793](https://github.com/dealii/dealii/pull/16299/commits/04cb793c0b43edf3da42cee66b0ca183cf001a79)

Reply:
[c9799dd](https://github.com/dealii/dealii/pull/16299/commits/c9799dd3d6597d402dd04a1c06e0beb9e882212e)